lib/daru/vector.rb

```diff
@@ -352,7 +352,14 @@ def == other
         if other.is_a?(Daru::Vector)
           mod.apply_vector_operator operator, self, other
         else
-          mod.apply_scalar_operator operator, @data,other
+          missing_undefined_indexes = nil_positions + nan_positions
+          if operator != :== && operator != :!= && missing_undefined_indexes.size
+            data_dup = @data.dup
+            missing_undefined_indexes.each { |i| data_dup.delete_at(i) }
+            mod.apply_scalar_operator operator, data_dup, other
+          else
+            mod.apply_scalar_operator operator, data, other
+          end
```

Comment:
@zverok , according to this link : http://ruby-doc.org/core-2.0.0/NilClass.html 

the problem is in `nil` (it can handle the `NaN`) . `nil` is `0` so should we compare it as `0` ?